File: src/Array/MinimumTimeDifference/MinimumTimeDifference.java
class Solution {
    public int findMinDifference(List<String> timePoints) {
        // Convert time to minutes
        int maxMinutes = 0;
        Set<Integer> set = new HashSet<>();
        for (String time : timePoints) {
            Integer minutes = convertToMinutes(time);
            maxMinutes = Math.max(minutes, maxMinutes);
            if (!set.add(minutes))
                return 0;
        }
        // From max minutes to 0 to find the min time difference
        Integer preMinutes = null, minDifference = Integer.MAX_VALUE;
        for (int curMinutes = maxMinutes; curMinutes >= 0; curMinutes--) {
            if (set.contains(curMinutes)) {
                if (preMinutes != null) {
                    minDifference = Math.min(minDifference, preMinutes - curMinutes);
                    minDifference = Math.min(minDifference, (curMinutes + 24 * 60) - preMinutes);
                }
                preMinutes = curMinutes;
            }
        }
        // Final Check
        minDifference = Math.min(minDifference, maxMinutes - preMinutes);
        minDifference = Math.min(minDifference, (preMinutes + 24 * 60) - maxMinutes);
        return minDifference;
    }

    private Integer convertToMinutes(String time) {
        String[] timeArr = time.split(":");// timeArr[0] = hour and timeArr[1] = minutes
        return Integer.parseInt(timeArr[0]) * 60 + Integer.parseInt(timeArr[1]);
    }
}

File: src/Array/MinimumTimeDifference/MinimumTimeDifference.py
class Solution:
    def findMinDifference(self, timePoints: List[str]) -> int:

        def convert_to_minutes(time: str) -> int:
            time_arr = time.split(":")
            return int(time_arr[0]) * 60 + int(time_arr[1])

        times = set()
        max_minutes = 0

        for time in timePoints:
            minutes = convert_to_minutes(time)
            if minutes in times:
                return 0
            else:
                times.add(minutes)
                max_minutes = max(minutes, max_minutes)

        pre_minutes = None
        min_diff = one_day_in_minutes = 24 * 60

        for cur_minutes in range(max_minutes, -1, -1):
            if cur_minutes in times:
                if pre_minutes is not None:
                    min_diff = min(min_diff, pre_minutes - cur_minutes,
                                   (cur_minutes + one_day_in_minutes) - pre_minutes)
                pre_minutes = cur_minutes

        # Compare min minutes with max minutes
        min_diff = min(
            min_diff, (pre_minutes + one_day_in_minutes) - max_minutes)

        return min_diff
